src/fighter_state.rs

```diff
@@ -0,0 +1,747 @@
+                    Throwing,
+                    Throwing::PRIORITY,
+                    true,
+                ));
+            } else {
+                transition_intents.push_back(StateTransition::new(
+                    Grabbing,
+                    Grabbing::PRIORITY,
+                    true,
+                ));
+            }
+        }
+
+        // Trigger movement
+        if action_state.pressed(PlayerAction::Move) {
+            let dual_axis = action_state.clamped_axis_pair(PlayerAction::Move).unwrap();
+            let direction = dual_axis.xy();
+
+            transition_intents.push_back(StateTransition::new(
+                Moving {
+                    velocity: direction * stats.movement_speed,
+                },
+                Moving::PRIORITY,
+                false,
+            ));
+        }
+    }
+}
+
+/// Look for attacks that have contacted a figher and knock them back
+fn collect_attack_knockbacks(
+    mut fighters: Query<&mut StateTransitionIntents, With<Handle<FighterMeta>>>,
+    mut damage_events: EventReader<DamageEvent>,
+) {
+    for event in damage_events.iter() {
+        // If the damaged entity was a fighter
+        if let Ok(mut transition_intents) = fighters.get_mut(event.damaged_entity) {
+            // Trigger knock back
+            transition_intents.push_back(StateTransition::new(
+                KnockedBack {
+                    velocity: event.damage_velocity,
+                    timer: Timer::from_seconds(0.18, false),
+                },
+                KnockedBack::PRIORITY,
+                false,
+            ));
+        }
+    }
+}
+
+/// Look for fighters with their health depleated and transition them to dying state
+fn collect_fighter_eliminations(
+    mut fighters: Query<(&Health, &mut StateTransitionIntents), With<Handle<FighterMeta>>>,
+) {
+    for (health, mut transition_intents) in &mut fighters {
+        // If the fighter health is depleted
+        if **health <= 0 {
+            // Transition to dying state
+            transition_intents.push_back(StateTransition::new(Dying, Dying::PRIORITY, false));
+        }
+    }
+}
+
+//
+// Transition states systems
+//
+
+/// Initiate any transitions from the idling state
+fn transition_from_idle(
+    mut commands: Commands,
+    mut fighters: Query<(Entity, &mut StateTransitionIntents), With<Idling>>,
+) {
+    for (entity, mut transition_intents) in &mut fighters {
+        // Transition to higher priority states
+        transition_intents.transition_to_higher_priority_states::<Idling>(
+            entity,
+            Idling::PRIORITY,
+            &mut commands,
+        );
+    }
+}
+
+// Initiate any transitions from the flopping state
+fn transition_from_attacking(
+    mut commands: Commands,
+    mut fighters: Query<(Entity, &mut StateTransitionIntents, &Attacking)>,
+) {
+    'entity: for (entity, mut transition_intents, flopping) in &mut fighters {
```

Comment:
I am unfamiliar with the `'entity:` syntax here. Is it a lifetime thing?

Reply:
Nope, that's actually a [loop label](https://doc.rust-lang.org/stable/reference/expressions/loop-expr.html?highlight=label#loop-labels).

Look down below where we say `continue 'entity`. It allows us to continue the entity loop, and would help distinguish which loop you were `continue`-ing, if you needed to `continue` from inside a nested loop.

I needed it at one point, but apparently I took out whatever nested loop I had and it's flat now, so that isn't necessary anymore. I can take it out.

It's actually a really nice feature I've never seen in another language, but I guess we don't need it here anymore. :)

Reply:
Oh very interesting. Thanks for explaining, I'm happy to know about that.
If we don't need it here definitely remove it.